sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/// <reference lib="esnext.asynciterable" />
+
+import {
+  operationOptionsToRequestOptionsBase,
+  TokenCredential,
+  isTokenCredential,
+  signingPolicy,
+  createPipelineFromOptions
+} from "@azure/core-http";
+import { PageSettings, PagedAsyncIterableIterator } from "@azure/core-paging";
+
+import { challengeBasedAuthenticationPolicy } from "../../keyvault-common/src";
+import {
+  RoleAssignmentsCreateResponse,
+  RoleAssignmentsDeleteResponse,
+  RoleAssignmentsListForScopeOptionalParams
+} from "./generated/models";
+
+import {
+  CreateRoleAssignmentOptions,
+  KeyVaultRoleAssignment,
+  AccessControlClientOptions,
+  RoleAssignmentScope,
+  DeleteRoleAssignmentOptions,
+  ListRoleAssignmentsOptions,
+  ListRoleDefinitionsOptions,
+  KeyVaultRoleDefinition
+} from "./accessControlModels";
+import { SDK_VERSION, LATEST_API_VERSION } from "./constants";
+import { KeyVaultClient } from "./generated/keyVaultClient";
+import { createSpan, setParentSpan } from "./tracing";
+import { mappings } from "./mappings";
+import { logger } from "./log";
+
+export class AccessControlClient {
```

Comment:
We may want to consider a more specific name such as `KeyVaultAccessControlClient`

Reply:
cc: @heaths

Reply:
Yes, it should be KeyVaultAccessControlClient because we will likely be having a more generic data-plane (and possibly management-plane) AccessControlClient. We want to make sure that it's scoped to Key Vault.

Reply:
Perfect! Thank you

Reply:
Addressed here: https://github.com/Azure/azure-sdk-for-js/pull/10815/commits/b7d3bf58cb6442a2e87b7ec2c780ab6181844474

Please review again!